src/DotNetWorker.Core/Converters/MemoryConverter.cs

```diff
@@ -3,33 +3,33 @@
 
 using System;
 using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace Microsoft.Azure.Functions.Worker.Converters
 {
-    internal class MemoryConverter : IConverter
+    internal class MemoryConverter : IInputConverter
     {
-        public bool TryConvert(ConverterContext context, out object? target)
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
         {
-            target = default;
-
             if (context.Source is not ReadOnlyMemory<byte> sourceMemory)
             {
-                return false;
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
```

Comment:
Do we return unhandled here because it can't be handled? I feel like `Unhandled` is being used for two different scenarios and it's a little confusing:

1. We can't handle the request
1. We can handle the request but this request is currently unhandled?

If it's #1 that we mean here, it might be worth renaming that so it's easier to understand i.e. "unhandleable", alternativetly (if this is causing so much confusion) we remove the `handled` property respond with ConverstionResult.Error instead?

I think this kind of goes back to the other thread we have in this PR about states and IsHandled & IsSuccessful